Controller/TemplateController.php

```diff
@@ -15,39 +15,28 @@
 use Symfony\Component\HttpFoundation\JsonResponse;
 use Symfony\Component\HttpFoundation\Request;
 use Symfony\Component\HttpFoundation\Response;
-use Symfony\Component\Serializer\Serializer;
-use Symfony\Component\Serializer\Encoder\JsonEncoder;
-use Symfony\Component\Serializer\Normalizer\GetSetMethodNormalizer;
-use CampaignChain\CoreBundle\Entity\Module;
 use CampaignChain\CoreBundle\Entity\Action;
 
+/**
+ * Class TemplateController
+ * @package CampaignChain\Campaign\TemplateBundle\Controller
+ */
 class TemplateController extends Controller
 {
     const CAMPAIGN_DISPLAY_NAME = "Campaign Template";
     const BUNDLE_NAME = 'campaignchain/campaign-template';
     const MODULE_IDENTIFIER = 'campaignchain-template';
     const TRIGGER_HOOK = 'campaignchain-timespan';
 
-    public function indexAction(){
+    public function indexAction()
+    {
         // Get the campaign templates
-        $qb = $this->getDoctrine()->getEntityManager()->createQueryBuilder();
-        $qb->select('c')
-            ->from('CampaignChain\CoreBundle\Entity\Campaign', 'c')
-            ->from('CampaignChain\CoreBundle\Entity\Module', 'm')
-            ->from('CampaignChain\CoreBundle\Entity\Bundle', 'b')
-            ->where('b.name = :bundleName')
-            ->andWhere('m.identifier = :moduleIdentifier')
-            ->andWhere('m.id = c.campaignModule')
-            ->setParameter('bundleName', static::BUNDLE_NAME)
-            ->setParameter('moduleIdentifier', static::MODULE_IDENTIFIER)
-            ->orderBy('c.name', 'ASC');
-        $query = $qb->getQuery();
-        $repository_campaigns = $query->getResult();
+        $repository_campaigns = $this->getDoctrine()->getRepository('CampaignChainCoreBundle:Campaign')->getCampaignsByModule(static::MODULE_IDENTIFIER);
```

Comment:
Understood. Thanks for clarifying. Then, the initial SQL was wrong. Only the combination of bundle name plus module identifier allows to find the exact module. The reason being, that if we used only the module identifier, there could be multiple results, if different bundles included modules with the same name. For example, a bundle acme/campaign-template might include a module campaignchain-template to extend the functionality of the module campaignchain-template in campaignchain/campaign-template.

Could you please fix this accordingly within the new code?